csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -1653,6 +1657,214 @@ private void TestInferenceSessionWithByteArray()
             }
         }
 
+        void TestCPUAllocator(InferenceSession session)
+        {
+            int device_id = 0;
+            using (var info_cpu = new OrtMemoryInfo(OrtMemoryInfo.allocatorCPU, OrtAllocatorType.ArenaAllocator, device_id, OrtMemType.Default))
+            {
+                Assert.NotEqual(info_cpu.Pointer, IntPtr.Zero);
+                Assert.Equal("Cpu", info_cpu.Name);
+                Assert.Equal(device_id, info_cpu.Id);
+                Assert.Equal(OrtAllocatorType.ArenaAllocator, info_cpu.GetAllocatorType());
+                Assert.Equal(OrtMemType.Default, info_cpu.GetMemoryType());
+
+                using (var allocator = new OrtAllocator(session, info_cpu))
+                {
+                    Assert.NotEqual(allocator.Pointer, IntPtr.Zero);
+                    var alloc_info = allocator.Info;
+                    Assert.True(info_cpu.CompareMemoryInfo(alloc_info));
+
+                    uint size = 1024;
+                    OrtMemoryAllocation chunk = allocator.Allocate(size);
+                    Assert.NotEqual(chunk.Pointer, IntPtr.Zero);
+                    Assert.Equal(chunk.Size, size);
+                    Assert.True(chunk.Info.CompareMemoryInfo(alloc_info));
+                    chunk.Dispose();
+                    alloc_info.Dispose();
+                }
+            }
+        }
+
+#if USE_CUDA
+        void TestCUDAAllocator(InferenceSession session)
```

Comment:
Fact ?

Reply:
It is helper, I should rename it

---
In reply to: [463255131](https://github.com/microsoft/onnxruntime/pull/4646#discussion_r463255131) [](ancestors = 463255131)